cmd/out/main.go

```diff
@@ -49,6 +53,27 @@ func main() {
 		blobPath := filepath.Dir(outRequest.Source.Regexp)
 		blobBaseName := filepath.Base(outRequest.Params.File)
 		blobName = filepath.Join(blobPath, blobBaseName)
+
+		matcher, err := regexp.Compile(outRequest.Source.Regexp)
+		if err != nil {
+			log.Fatal("failed to compile source configuration regex: ", err)
+		}
+
+		matches := matcher.FindStringSubmatch(blobName)
```

Comment:
The `blobName` pulls from the params and isn't the actual path, which has the version in it. I think this code needs to move lower down to after the `out.UploadFileToBlobstore` call and matches on the `path`.

Noticed this by running the `./acceptance` tests. Specifically [this test](https://github.com/pivotal-cf/azure-blobstore-resource/blob/master/acceptance/out_test.go#L149) if you'd like one to focus. Unfortunately don't have GitHub PR actions to run the tests so you'll have to run this locally. It should be fairly straight forward though, you will need a storage account name and the storage account key the tests can use to upload/download files to/from Azure. More info [here](https://github.com/pivotal-cf/azure-blobstore-resource/blob/master/acceptance/README.md). Happy to answer any questions if you get stuck.

Reply:
Thanks for the feedback, Christian! I'd taken a look at the unit tests but completely missed the `./acceptance` directory, I'll have a look at enhancing the test cases to make sure we check for the version on the output and get this fixed up.